include/versioningCompiler/Compiler.hpp

```diff
@@ -97,25 +97,25 @@ class Compiler
   virtual std::string generateBin(const std::vector<std::string> &src,
                                   const std::vector<std::string> &func,
                                   const std::string &versionID,
-                                  const opt_list_t options) const = 0;
+                                  const opt_list_t options) = 0;
 
-/** \brief Opens the binary shared object, stores in *handler the reference to
+     /** \brief Opens the binary shared object, stores in *handler the reference to
  * the open shared object, and loads the symbol relative to the given function.
  *
  * Returns the loaded function pointer on success. nullptr otherwise.
  * Output of this method is supposed to reinterpreted by the caller.
  */
-  std::vector<void*> loadSymbols(const std::string &bin,
+ virtual std::vector<void*> loadSymbols(std::string &bin,
```

Comment:
why removing `const` here?

Reply:
`std::string &bin` was being used as a handler for the JITCompiler implementation. I have just changed JITCompiler implementation in order to keep this as const.